### PR TITLE
Issue #15: Improve geocoder_drush_backfill(). (7.x-1.x)

### LINKED ISSUE #15: Fix `geocoder_drush_backfill()` in 7.x-1.x
The command callback `geocoder_drush_backfill()` for drush command `drush geocoder-backfill` currently has some problems:
- Only works for nodes, not other entity types.
- Overall the structure could be improved.
- More options.

I am going to make a PR.

I realize the code here on github is a bit behind, the PR will be on top of latest 7.x-1.x from drupal.org.

### PR DESCRIPTION
**Pull request by donquixote**
Fixes #15

### CODE REVIEW

**Comment by donquixote** [comment]
I think the patches from the drupal.org issue also have some interesting points, we might want to integrate some of this into the PR.
https://www.drupal.org/project/geocoder/issues/2077317#comment-9260663

I originally started to code this PR from scratch, and only later found the drupal.org issue.

**Comment by donquixote** [comment]
One thing I wonder about is the per-language and per-delta stuff.
E.g. if the source field has an item in English and one in French, and the (geocoded) destination field only has the English item, then does it need recalculation?

Perhaps this depends on the geocoder widget settings, which kind of mapping is defined there?

Btw I don't use `field_get_items()` because this gives items in a specific language.
I think the drush command should cover all languages at once, not just the "active" one (which in drush command is probably English).

I also don't query the entity table directly. I think it is smarter to query the source field table, because we are only interested in items that have data in the source field.

Also we don't have to loop through all entity types, better loop through all field instances.

**Comment by donquixote** [comment]
The drupal.org issue has different names for arguments.

**Comment by donquixote** [comment]
And @drupol can you push latest 7.x-1.x to github?

**Comment by drupol** [comment]
@donquixote Done! Waiting for your rebase and then review/merge :-)

**Comment by donquixote** [comment]
@drupol There wasn't really anything to rebase. I just had to remind github to do refresh the PR.
So.. I just did `git commit --amend`, changed nothing, and did a `git push --force-with-lease donquixote 7.x-1.x--15--geocoder_drush_backfill`.

File: geocoder.drush.inc [hunk 1]
```diff
@@ -12,57 +12,115 @@ function geocoder_drush_command() {
...
   );
 
   return $items;
 }
 
+/**
+ * Drush command callback for 'geocoder-backfill'.
+ */
 function geocoder_drush_backfill() {
+
```
Comment (donquixote) [inline geocoder.drush.inc]: Oh dear. I expect you will not like this. I didn't even realize it :)

Comment (drupol) [reply geocoder.drush.inc]: Exactly :-)

File: geocoder.drush.inc [hunk 2]
```diff
@@ -12,57 +12,115 @@ function geocoder_drush_command() {
...
-                if ($field_info['type'] == 'location') {
-                  if ($force_reload || empty($items['latitude'])) node_save($entity); //TODO: fix for all entities
-                }
-                if ($field_info['type'] == 'geolocation') {
-                  if ($force_reload || empty($items['lat'])) node_save($entity); //TODO: fix for all entities
-                }
-              }
-            }
-          }
+        if ([] !== $etids_orphan) {
```
Comment (drupol) [inline geocoder.drush.inc]: For consistency, I think we should use the old array notation here.

**Comment by donquixote** [comment]
You will see the light one day.

**Comment by donquixote** [comment]
I fixed the two points you mentioned.

**Pull request merged**